### PR TITLE
How to delete all tickets in OM

### PR DESCRIPTION
**Pull request by yfei1**
Resolved [open-match#1136](https://github.com/googleforgames/open-match/issues/1136)

image

### CODE REVIEW

File: site/content/en/docs/FAQ/delete_all.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+fc, err := grpc.Dial(queryServiceAddr, grpc.WithInsecure())
+if err != nil {
+    log.Fatalf("Failed to connect to Open Match, got %s", err.Error())
+}
+defer fc.Close()
+
+f := pb.NewQueryServiceClient(conn)
+
+// an empty pool returns all ticket ids in Open Match
+stream, err := qc.QueryTicketIds(context.Background(), &pb.QueryTicketIdsRequest{Pool: nil})
```
Comment (Laremere) [inline site/content/en/docs/FAQ/delete_all.md]: This will return an error.

File: site/content/en/docs/FAQ/delete_all.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+for {
```
Comment (Laremere) [inline site/content/en/docs/FAQ/delete_all.md]: Probably want to finish getting all ids before deleting.

File: site/content/en/docs/FAQ/delete_all.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    }
+
+    for _, id := range resp.GetIds() {
+        _, err = f.DeleteTicket(context.Background(), &pb.DeleteTicketRequest{TicketId: id})
+        if err != nil {
+            // process the error
+        }
+    }
+}
+```
```
Comment (Laremere) [inline site/content/en/docs/FAQ/delete_all.md]: How does this handle pending tickets or tickets that are already assigned?

Comment (yfei1) [reply site/content/en/docs/FAQ/delete_all.md]: So what is the agreement on #1136? I thought this doc was intended to document how to delete all tickets in OM?

Comment (sawagh) [reply site/content/en/docs/FAQ/delete_all.md]: More than the documentation itself, I am concerned about the implementation for: 
https://github.com/googleforgames/open-match/pull/1151/

Looks like that goes through query service - so I assume you will only get Active IDs? So is this statement on that API proto accurate:
_If the Pool contains no Filters, QueryTicketIds will return all TicketIDs in the state storage._ ?

The primary user scenario here was to have an ability to cleanup Open Match state storage - and if there is a simpler way to do that, we should consider that. Providing a state specific API may be cleaner (get all pending / get all assigned / get all in all states) but thats a lower priority.

Also, we dont expect this to be called at scale - so going through query cache for this is not a requirement.

**Comment by yfei1** [comment]
Closing this as we gonna move this issue to v1.0

**Pull request closed without merging**